### PR TITLE
Refactor search endpoint

### PR DESCRIPTION
**Pull request by mario-galic**
Use `Option.exists` instad of `Option.getOrElse`

### CODE REVIEW

File: app/controllers/UsersController.scala [hunk 1]
```diff
@@ -29,21 +29,26 @@ class UsersController @Inject() (
...
-    }
-    else if (limit.getOrElse(0) < 0) {
-      Future.successful(BadRequest(ApiError("limit must be a positive integer")))
-    }
-    else if (query.length < MinimumQueryLength) {
-      Future.successful(BadRequest(ApiError(s"query must be a minimum of $MinimumQueryLength characters")))
-    }
-    else {
+    def validateSearchQuery() =
+      if (!offset.forall(_ >= 0))
```
Comment (jfsoul) [inline app/controllers/UsersController.scala]: Sorry maybe my suggestion before was not the best solution: is this neater as `if (offset.exists(_ < 0))`?  I  think that is the same thing but check my logic!

Comment (mario-galic) [inline app/controllers/UsersController.scala]: You are right, `offset.exists` is better. 

What do you think about these inner methods`validateSearchQuery` and `doSearch`? Is it clearer or should I leave it as it was?

Comment (jfsoul) [inline app/controllers/UsersController.scala]: I am not sure about it, I think I don't like the wrapping with `Left` and `Right(())` - it indicates there should be a more natural way to write this.

Comment (mario-galic) [inline app/controllers/UsersController.scala]: See https://github.com/guardian/identity-admin-api/pull/137 for potentially more natural way of writing this.

**Pull request merged**